packages/form-js-viewer/src/render/components/form-fields/parts/Timepicker.js

```diff
@@ -0,0 +1,146 @@
+
+    const adjustedTime = isNullTime ? null : time - (time % timeInterval);
+    setRawValue(isNullTime ? '' : formatTime(use24h, adjustedTime));
+    setTime(adjustedTime);
+  }, [ time, setTime, use24h, timeInterval ]);
+
+  const propagateRawToMinute = useCallback((newRawValue) => {
+
+    const localRawValue = newRawValue || rawValue;
+
+    // If no raw value exists, set the minute to null
+    if (!localRawValue) {
+      setTime(null);
+      return;
+    }
+
+    const minutes = parseInputTime(localRawValue);
+
+    // If raw string couldn't be parsed, clean everything up
+    if (!isNumber(minutes)) {
+      setRawValue('');
+      setTime(null);
+      return;
+    }
+
+    // Enforce the minutes to match the timeInterval
+    const correctedMinutes = minutes - (minutes % timeInterval);
+
+    // Enforce the raw text to be formatted properly
+    setRawValue(formatTime(use24h, correctedMinutes));
+    setTime(correctedMinutes);
+
+  }, [ rawValue, timeInterval, use24h, setTime ]);
+
+  const timeOptions = useMemo(() => {
+
+    const minutesInDay = 24 * 60;
+    const intervalCount = Math.floor(minutesInDay / timeInterval);
+    return [ ...Array(intervalCount).keys() ].map(intervalIndex => formatTime(use24h, intervalIndex * timeInterval));
+
+  }, [ timeInterval, use24h ]);
+
+  const initialFocusIndex = useMemo(() => {
+
+    // If there are no options, there will not be any focusing
+    if (!timeOptions) return null;
+
+    // If there is a set minute value, we focus it in the dropdown
+    if (time) return time / timeInterval;
+
+    // If there is no set value, simply focus the middle of the dropdown (12:00)
+    return Math.floor(timeOptions.length / 2);
+
+  }, [ time, timeInterval, timeOptions ]);
+
+  const onInputKeyDown = (e) => {
+    switch (e.key) {
+    case 'ArrowUp':
+      e.preventDefault();
+      break;
+    case 'ArrowDown':
+      useDropdown && setDropdownIsOpen(true);
+      e.preventDefault();
+      break;
+    case 'Escape':
+      useDropdown && setDropdownIsOpen(false);
+      break;
+    case 'Enter':
+      !dropdownIsOpen && propagateRawToMinute();
+      break;
+    }
+  };
+
+  const onInputBlur = (e) => {
+    setDropdownIsOpen(false);
+    propagateRawToMinute();
+  };
+
+  const onDropdownValueSelected = (o) => {
+    setDropdownIsOpen(false);
+    propagateRawToMinute(o);
+  };
+
+  return <InputAdorner
+    pre={ <ClockIcon /> }
+    inputRef={ timeInputRef }
+    disabled={ disabled }>
+    <div style={ { width: '100%' } } class="fjs-timepicker-anchor">
+      <input ref={ timeInputRef }
+        type="text"
+        id={ `${prefixId(id, formId)}--time` }
+        class="fjs-input"
+        value={ rawValue }
+        disabled={ disabled }
+        placeholder={ use24h ? 'HH:MM' : 'HH:MM ?M' }
+        autoComplete="false"
+        onFocus={ () => useDropdown && setDropdownIsOpen(true) }
+        onClick={ () => useDropdown && setDropdownIsOpen(true) }
+
+        // @ts-ignore
```

Comment:
What happens if we remove `@ts-ignore`?

Reply:
`e.target.value` is apparently not guaranteed to exist, this is because the fact that we're defining an event on an input field isn't part of the ts context. This one I cannot remove.